traffic_ops/traffic_ops_golang/region/regions_test.go

```diff
@@ -109,3 +108,30 @@ func TestInterfaces(t *testing.T) {
 		t.Errorf("Region must be Identifier")
 	}
 }
+func TestValidation(t *testing.T) {
+	testRegion := tc.Region{
+		DivisionName: "west",
+		ID:           1,
+		Name:         "region1",
+		LastUpdated:  tc.TimeNoMod{Time: time.Now()},
+	}
+	testTORegion := TORegion{Region: testRegion}
+	errs := test.SortErrors(test.SplitErrors(testTORegion.Validate()))
+
+	if len(errs) > 0 {
+		t.Errorf(`expected no errors,  got %v`, errs)
```

Comment:
This expects no errors because you provided `DivisionName`, but the validation requires `Division` *and* `DivisionName` as it's currently written. Which is why the test fails.

Reply:
Yes, had changes to do that but pivoted because it was pretty late

Test fixture updated to include a division, error message updated

Reply:
What is the intended functionality here for this validation? Should both Division and DivisionName be required?

Reply:
No, I don't think so. I think only the ID should be required, as the docs say:

> _"The only “division” key that actually matters in the request body is `division`; `divisionName` is not validated and has no effect - particularly not the effect of re-naming the division - beyond changing the name in the API response to this request. Subsequent requests will reveal the true name of the division. Note that if `divisionName` is not present in the request body it will be null in the response, but again further requests will show the true division name (provided it has been assigned to a division)._"

Reply:
Removed divisionName from the validation, updated test slightly